application-task-default/src/main/java/com/xwiki/task/internal/notifications/TaskChangedEventFactory.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package com.xwiki.task.internal.notifications;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.xwiki.model.reference.LocalDocumentReference;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.objects.DateProperty;
+import com.xpn.xwiki.objects.LargeStringProperty;
+import com.xpn.xwiki.objects.PropertyInterface;
+import com.xpn.xwiki.objects.StringProperty;
+
+/**
+ * Create the necessary {@link TaskChangedEvent}s for a task update.
+ * 
+ * @version $Id$
+ * @since 3.7
+ */
+public final class TaskChangedEventFactory
```

Comment:
You should have it as a Singleton [Component](TaskChangedEventFactory) that can be injected. Then this will have public methods that can be used